src/Psl/Iter/times.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Iter;
+
+use Psl;
+
+/**
+ * Creates an iterator containing all numbers between the start and end value
+ * (inclusive) with a certain step.
+ *
+ * Examples:
+ *
+ *     Iter\times(5, Fun\Identity())
+ *     => Iter(1, 2, 3, 4, 5)
+ *
+ * @psalm-template T
+ *
+ * @psalm-param int $number How many times should the factory be executed?
+ * @psalm-param (callable(int): T) $factory
+ *
+ * @psalm-return Iterator<int, T>
+ *
+ * @throws Psl\Exception\InvariantViolationException If $number < 1
+ */
+function times(int $number, callable $factory): Iterator
```

Comment:
I think for the sake of consistency with `Str\repeat` and `Iter\repeat`, the `$number` argument must be last.

`times` doesn't really hold for me here, as it can be intercepted as `Iter\repeat`, what do you think about `reproduce` instead ? 

```suggestion
function reproduce(callable $factory, ?int $number = null): Iterator
```

( when a null argument is given, this will produce `Math\INFINITY` items, similar to `Iter\repeat` )

Reply:
Sounds good! The INFINITY thing is nice! makes sense